src/test/java/io/neonbee/health/AbstractHealthCheckTest.java

```diff
@@ -79,12 +98,74 @@ void testRegisterNodeCheck(VertxTestContext testContext) throws HealthCheckExcep
 
         hc.register(registry).onComplete(testContext.succeeding(check -> testContext.verify(() -> {
             assertThat(check.isGlobal()).isFalse();
-            verify(registry).registerNodeCheck(eq(DUMMY_ID), eq(DEFAULT_RETENTION_TIME), any(), eq(new JsonObject()));
+            verify(registry).registerNodeCheck(eq(DUMMY_ID), eq(DEFAULT_RETENTION_TIME), any(), eq(HEALTH_CONFIG));
+            testContext.completeNow();
+        })));
+    }
+
+    @Test
+    @Timeout(value = 2, timeUnit = TimeUnit.SECONDS)
+    @DisplayName("can merge a config from file in config directory with the default config")
+    void testMergeConfigs(VertxTestContext testContext) {
+        when(fsMock.readFile(anyString())).thenReturn(
+                succeededFuture(new JsonObject().put("enabled", false).put("new-random-key", "something").toBuffer()));
+        when(vertxMock.fileSystem()).thenReturn(fsMock);
+
+        hc.mergeHealthCheckConfig().onComplete(testContext.succeeding(config -> testContext.verify(() -> {
+            String path =
+                    NeonBee.get(vertxMock).getOptions().getConfigDirectory().resolve(hc.getClass().getName()) + ".yaml";
+            verify(fsMock).readFile(eq(path));
+            assertThat(config.getBoolean("enabled")).isFalse();
+            assertThat(config.getString("new-random-key")).isEqualTo("something");
+            testContext.completeNow();
+        })));
+    }
+
+    static class CustomConfigArgumentProvider implements ArgumentsProvider {
+
+        @Override
+        public Stream<? extends Arguments> provideArguments(ExtensionContext context) {
+            return Stream.of(Arguments.of(succeededFuture(new JsonObject().toBuffer()), "config from file is empty"),
+                    Arguments.of(succeededFuture(Buffer.buffer()), "config from file is an empty buffer"),
+                    Arguments.of(failedFuture("oops"), "retrieving config fails"));
+        }
+    }
+
+    @ParameterizedTest(name = "{index} => {1}")
+    @ArgumentsSource(CustomConfigArgumentProvider.class)
```

Comment:
Why you do this via a provider class instead of a provider method?

Reply:
No specific reason :)